### DISCUSSION TITLE
Canceling in-flight requests

### DISCUSSION THREAD

**Discussion by ymc9** [jycouet/kitql#201] [Q&A]
Hi there,

I'm still on kitql 0.6.7 and having an issue with rendering wrong data. In my svelte routes, I use .query() to trigger GQL queries and use stores generated by kitql to render. This is convenient.

The problem is that if I switch back and forth in routes, the same GQL queries can be triggered with different parameters. Since the store generated for a query is global, it can be loaded with non-deterministic data (depending on which query returned last).

I thought I needed to cancel in-flight requests before triggering new ones, but haven't figured out how to do that. Comments are appreciated. Thanks!

**Comment by ymc9** [comment]
Actually I feel there's more problem with the global store. Is it true that I need to reset store manually before triggering a new query() call? Otherwise, when the query is still in-flight, the store still serves stale data.

Should the store be somehow keyed by query input, like cache is?

**Reply by jycouet** [reply]
This was one of the limitations of the KitQL Client.
Also, prefetching was not supported.
`Global` can be very powerful in some cases and limiting in others.

We are taking a way better approach with the new version and actually Houdini.
Any constraints why not upgrading? Could I help you? 
Do not hesitate to contact me 👍

**Reply by ymc9** [reply]
Thanks for answering @jycouet. I would love to upgrade but just haven’t got enough free time to do it 😄. I’ll checkout Houdini doc to understand how the new approach works.
